File: Services/TaskService/TaskService.Application/UseCases/ResourceUseCases/UpdateResource/UpdateResourceCommand.cs
using MediatR;
using Microsoft.AspNetCore.Http;
using TaskService.Domain.Enums;

namespace TaskService.Application.UseCases
{
    public record UpdateResourceCommand(
        string? username,
        int? companyId,
        int? resourceId,
        string? data,
        IFormFile? imageFile,
        ResourceType? type
    ) : IRequest<Unit>;
}

File: FrontendApp/src/app/models/commands/taskservice/update-resource.command.ts
import { ResourceType } from "../../../enums/resource-type.enum";

export interface UpdateResourceCommand {
  username: string | null;
  companyId: number | null;
  resourceId: number | null;
  data: string | null;
  imageFile: File | null;
  type: ResourceType | null;
}
